### ISSUE TITLE
Feature request: closing tabs in `T` dialog

### ISSUE THREAD

**Issue by agzam** [philc/vimium#2078]
Would be awesome to be able to close any tab right in `T` window

**Comment by gqqnbig** [comment]
For your information, it's a duplicate of #1735 and #1863. They were closed as lack of interest, though I personally think it's useful to me.

**Comment by innaterebel** [comment]
Not sure why #1735 was evaluated as lack of interest, given all the comments there seem to support the idea. Should we post `+1`s to show interest?:wink:

**Comment by afilp** [comment]
Can we please reconsider this? It is **very** useful and it can make us remove other extensions for tabs management.

I have the following idea for the key combination problem:

If the input does not have a **selection** of text, then `SHIFT-DEL` closes the focused tab (otherwise it does as expecting, deleting the selected text):

![image](https://user-images.githubusercontent.com/7850073/51386425-04d32180-1b2b-11e9-90c0-3657b731d87d.png)

So, by simply pressing `SHIFT-DEL`, the tab closes and the next tab is focused in the list.

And so on.

I believe this (or any other variation you can think of) is pretty fast and does not seem to interfere with other shortcut actions.

What do you think?

**Comment by smblott-github** [comment]
@afilp I think this is reasonable, but I guess it would have to work for bookmarks and history too.

**Comment by afilp** [comment]
@smblott-github Yes, it can do the same, maybe with an alert warning in the case of bookmarks deletion.

Also, for safety in accidentally hitting the `SHIFT-DEL` we could require a **double press** of `SHIFT-DEL` (sequentially). Not sure if this is needed though.

**Comment by afilp** [comment]
@smblott-github I actually just had the need to delete a bookmark straight from the `b` dialog, see screenshot below:

![image](https://user-images.githubusercontent.com/7850073/52533963-e20be580-2d43-11e9-94b8-e4322b4177ec.png)

**Comment by philc** [comment]
I would find this very useful. Recently I somehow got a gigantic URL
recorded in my history, and it matches pretty much any query I type into
the open URL dialog, and that's the most convenient place to purge it from
my history.

I've also found that some low-signal webpages consistently rank highly in
my most frequent searches, and it would be convenient to just delete them
from the open URL dialog when I see them cluttering up the search results.

**Comment by balta2ar** [comment]
I also miss this feature very much. Could it be implemented for recent tabs only as a first step (without bookmarks and history for now)?

**Comment by IzakMarais** [comment]
Implementing this would allow me to remove another plug-in that I use for similar functionality.

**Comment by balta2ar** [comment]
FYI, this feature is implemented in Vimium-C alternative plugin, I switched to it, and experience has been great so far.

**Comment by gdh1995** [comment]
#3170 is also expecting this.

**Comment by gqqnbig** [comment]
@balta2ar  I switched to Vimium-C. I believe Vimium-C has more features than I can imagine, but it lacks documentations at least on this shift-delete feature.

**Comment by afilp** [comment]
@balta2ar How is it implemented there? I went to read the description there and could not find if this is supported.

**Comment by gdh1995** [comment]
@afilp you may try <kbd>Shift+Delete</kbd> to delete a tab/history item on Vomnibar.

**Comment by IzakMarais** [comment]
@aflip Yes it is not documented, but as Dahan says, Shift+Delete works.

**Comment by agzam** [comment]
I think `Shift+Delete` is not very convenient keybinding for Mac users, because it forces you to press `Fn+Shift+Delete` (without `Fn`, `Delete` works as `Backspace`). It makes it even harder if you use an external keyboard that doesn't even have `Fn` key, e.g.: Kinesis Advantage.

Is it possible somehow to rebind that action to be something else, maybe `Ctrl+D`?

**Comment by afilp** [comment]
@smblott-github Would you like to consider the "Shift-Del" (or ctrl+D for MAC) addition in the extension? Just for the loaded tabs for now, I have tons of loaded tabs and it would be really helpful to search, find those that I do not need and quickly press "Shift-Del" to close them.

**Comment by philc** [comment]
I'll take a look at this at some point; I would love this feature. The problem with ctrl-D is that it's bound on Mac as forward-delete, which I use all the time. As a mac user, I would be thrown off and annoyed if that was overwritten by Vimium.

**Comment by gdh1995** [comment]
We can add support for `mapkey` to Vomnibar, and <kbd>Shift+Delete</kbd> can be the default, so users can use whatever they like to do so.

There's a similar issues named `question: is it possible to remap vomnibar's shift+delete` in https://github.com/gdh1995/vimium-c/issues/112 .

I also recommend per-mode `mapkey` like `mapkey <m-d:o> <s-delete>` (map <kbd>Meta+D</kbd> to it): https://github.com/gdh1995/vimium-c/issues/110#issuecomment-580088981 .

**Comment by afilp** [comment]
@gdh1995 Is this useful feature implemented for bookmarks? Or maybe with another way? Thanks a lot!
https://github.com/philc/vimium/issues/2078#issuecomment-462130425

**Comment by gdh1995** [comment]
I think bookmarks are more important than history, so we should avoid mistaken deleting.



---Original---
From: ***@***.***&gt;
Date: Mon, Apr 17, 2023 22:42 PM
To: ***@***.***&gt;;
Cc: "Dahan ***@***.******@***.***&gt;;
Subject: Re: [philc/vimium] Feature request: closing tabs in `T` dialog(#2078)




 
@gdh1995 Is this useful feature implemented for bookmarks? Or maybe with another way? Thanks a lot!
 #2078 (comment)

**Comment by afilp** [comment]
In my use case, it is very difficult to press SHIFT-DEL by mistake. I find it very low risk.

Is it possible to have an option in settings? 

[ ] Allow SHIFT-DEL to delete the selected bookmark

The default can be false, so it is not a problem for those that do not want it. But it will be great for us that want it.

Thanks!

**Comment by kkuhle** [comment]
I've been trying to figure out how to accomplish this and finally got around to searching here to find that it can't be done. 

It would be really useful and give me more of the feel that I'm in a terminal/termux, but overall, just make it so much easier to manage my tabs without having a seperate extention for it or trying to manually navigate through tabs that I don't expect I need anymore. Here is what I currently use, but I'd love to remove an extra extension (even though it's a good one, but I prefer vimium):

https://github.com/philc/vimium/assets/11251306/ca97add6-4e07-4614-859d-3b2c23064013

Here is a similar situation (session which is more like a window than tab, but reminds this reminds me of what I want to do with vimium) in termux that provides a confirmation when closing when I have seperate windows:

https://github.com/philc/vimium/assets/11251306/e5a5966b-a3b8-486e-830f-b8b682b90fe3

**Comment by gdh1995** [comment]
@afilp Um, what I worried is such a case:
1. I want to delete some bookmarks, so type a few words to query it
2. I press `Up` / `Down` keys to select a suggestion
3. I press `Shift+Delete` and get the suggestion deleted
- 1. in my imagination, Vomnibar should refresh the list and the item will disappear
5. But, I suddenly notice that I deleted a wrong suggestion next/previous to a real target
6. There's no way to restore from the mistake ...

**Comment by aflip** [comment]
@afilp  and me @aflip are a word apart in username and so i keep getting updates from this repo. :) just saying hi and interesting project.
